tabular/src/autogluon/tabular/task/tabular_prediction/predictor_v2.py

```diff
@@ -0,0 +1,614 @@
+        try:
+            version_fit = predictor._learner.version
+        except:
+            version_fit = None
+        if version_fit is None:
+            version_fit = 'Unknown (Likely <=0.0.11)'
+        if version_inference != version_fit:
+            logger.warning('')
+            logger.warning('############################## WARNING ##############################')
+            logger.warning('WARNING: AutoGluon version differs from the version used during the original model fit! This may lead to instability and it is highly recommended the model be loaded with the exact AutoGluon version it was fit with.')
+            logger.warning(f'\tFit Version:     {version_fit}')
+            logger.warning(f'\tCurrent Version: {version_inference}')
+            logger.warning('############################## WARNING ##############################')
+            logger.warning('')
+
+        return predictor
+
+    @classmethod
+    def from_learner(cls, learner: AbstractLearner):
+        predictor = cls(label=learner.label, output_directory=learner.path)
+        predictor._set_post_fit_vars(learner=learner)
+        return predictor
+
+    @staticmethod
+    def _validate_init_kwargs(kwargs):
+        valid_kwargs = {
+            'learner_type',
+            'learner_kwargs',
+        }
+        invalid_keys = []
+        for key in kwargs:
+            if key not in valid_kwargs:
+                invalid_keys.append(key)
+        if invalid_keys:
+            raise ValueError(f'Invalid kwargs passed: {invalid_keys}\nValid kwargs: {list(valid_kwargs)}')
+
+    def _validate_fit_kwargs(self, kwargs):
+
+        # TODO:
+        #  Valid core_kwargs values:
+        #  ag_args, ag_args_fit, ag_args_ensemble, save_bagged_folds, stack_name, ensemble_type, name_suffix, time_limit
+        #  Valid aux_kwargs values:
+        #  name_suffix, time_limit, stack_name, aux_hyperparameters, ag_args, ag_args_ensemble
+
+        # TODO: Remove features from models option for fit_extra
+        # TODO: Constructor?
+        fit_kwargs_default = dict(
+            # data split / ensemble architecture kwargs -> Don't nest but have nested documentation -> Actually do nesting
+            holdout_frac=None,  # TODO: Potentially error if num_bag_folds is also specified
+            num_bag_folds=None,
+            auto_stack=False,
+
+            # other
+            unlabeled_data=None,
+        )
+
+        kwargs = self._validate_fit_extra_kwargs(kwargs, extra_valid_keys=list(fit_kwargs_default.keys()))
+
+        kwargs_sanitized = fit_kwargs_default.copy()
+        kwargs_sanitized.update(kwargs)
+
+        return kwargs_sanitized
+
+    def _validate_fit_extra_kwargs(self, kwargs, extra_valid_keys=None):
+        fit_extra_kwargs_default = dict(
+            # data split / ensemble architecture kwargs -> Don't nest but have nested documentation -> Actually do nesting
+            num_bag_sets=None,
+            num_stack_levels=None,
```

Comment:
num_bag_sets and num_stack_levels are free to change between fit calls, auto_stack is only available in the first fit call however.

`num_bag_folds` is very tricky, because it dictates `label_count_threshold`, which can't be changed. For example, if someone used `num_bag_folds=5` during first fit and a class only appeared 3 times and eval_metric=log_loss, it would be duplicated so it appeared 5 times so 1 example existed in each fold.

However, if they then called `num_bag_folds=10` in the 2nd call, there would be not enough samples for that class to exist in each fold, causing an exception. We could possibly avoid this by raising an exception pre-emptively if we determine a certain num_bag_folds value will cause this issue.

For now I've added a TODO to num_bag_folds to consider moving to fit_extra kwargs.